libbindgen/src/ir/item.rs

```diff
@@ -1214,8 +1221,45 @@ impl ItemCanonicalPath for Item {
             debug_assert!(is_alias, "How can this ever happen?");
             parent_path.pop().unwrap();
         }
-        parent_path.push(self.real_canonical_name(ctx, true, false));
+        parent_path.push(self.name().with_namespaces().get(ctx));
 
         parent_path
     }
 }
+
+/// Builder struct for naming variations, which hold inside different
+/// flags for naming options.
+#[derive(Debug)]
+pub struct NameOptions<'a> {
+    item: &'a Item,
+    with_namespaces: bool,
```

Comment:
Nitpick: `with_namespaces` doesn't clearly describe whether this is true if we want namespaces mangled into the resulting name, or the name within a namespace. We were using the latter before, and it doesn't look like that has changed, so I think this should be called `within_namespace` (open to bike shedding).